mlflow/gateway/providers/gemini.py

```diff
@@ -85,18 +87,57 @@ def chat_to_model(cls, payload, config):
 
         contents = []
         system_message = None
+
+        call_id_to_function_name_map = {}
+
         for message in payload["messages"]:
             role = message["role"]
 
-            if role == "assistant":
-                role = "model"
+            if role in ("user", "assistant"):
+                if role == "assistant":
+                    role = "model"
+
+                gemini_function_calls = []
```

Comment:
Is it possible that function calls are included in "user" messages?

Reply:
updated!